src/Microsoft.VisualStudio.Composition/Reflection/TypeRef.cs

```diff
@@ -242,7 +257,26 @@ public TypeRef MakeGenericTypeRef(ImmutableArray<TypeRef> genericTypeArguments)
 
             // We use the resolver parameter instead of the field here because this TypeRef instance
             // might have been constructed by TypeRef.Get(Type) and thus not have a resolver.
-            return new TypeRef(this.Resolver, this.AssemblyName, this.assemblyId, this.MetadataToken, this.FullName, this.TypeFlags, this.GenericTypeParameterCount, genericTypeArguments);
+            return new TypeRef(this.Resolver, this.AssemblyName, this.assemblyId, this.MetadataToken, this.FullName, this.TypeFlags, this.GenericTypeParameterCount, genericTypeArguments, this.ElementTypeRef);
+        }
+
+        public bool IsAssignableFrom(Type other)
+        {
+            if (other.EnumTypeBaseTypesAndInterfaces().Select(type => {
```

Comment:
We should probably avoid these delegates in these code paths to keep GC pressure down in a perf critical path. If you see it everywhere else nearby, then never mind. But I would expect we'd want to minimize here.